src/dcqc/reports.py

```diff
@@ -0,0 +1,40 @@
+import json
+from collections.abc import Iterator
+from typing import Any, Union
+
+from dcqc.mixins import SerializableMixin, SerializedObject
+from dcqc.utils import open_parent_fs
+
+
+class JsonReport:
+    url: str
+
+    def __init__(self, url: str, overwrite: bool = False) -> None:
+        self.url = url
+        self._fs, self._path = open_parent_fs(url)
+        if self._fs.exists(self._path) and not overwrite:
+            message = f"URL ({url}) already exists. Set `overwrite=True` to ignore."
+            raise FileExistsError(message)
+
+    def to_file(self, obj: Any):
+        with self._fs.open(self._path, "w") as outfile:
```

Comment:
I really like that upon closing done by the context manager, it uploads the file to Synapse if you specify a path in Synapse.  Incredible.